scenario/constraints/MigrationConstraint.ts

```diff
@@ -54,3 +52,23 @@ export class MigrationConstraint<T extends CometContext, R extends Requirements>
     return; // XXX
   }
 }
+
+export class VerifyMigrationConstraint<T extends CometContext, R extends Requirements> implements Constraint<T, R> {
+  async solve(requirements: R, context: T, world: World) {
+    const label = `[${world.base.name}] {VerifyMigrationConstraint}`;
+    return [
+      async function (ctx: T): Promise<T> {
+        for (const migration of ctx.migrations) {
+          // XXX does verify get the 'gov' deployment manager as well as the 'local' one?
+          if (migration.actions.verify) {
+            await migration.actions.verify(ctx.deploymentManager);
+            debug(`${label} Verified migration "${migration.name}"`);
+          }
+        }
+        return ctx;
+      }
+    ];
+  }
+
```

Comment:
I think you can add this to the `check` function of `MigrationConstraint` instead of creating a separate constraint.

Reply:
These have to run after the proposal executes not after its created

Reply:
Yeah, doesn't check happen after all constraints (e.g. proposal constraint) are run?

Reply:
Ah hm good point! Although this will all be affected as I split out static constraints, so I don't feel too bad having added the extra constraint, maybe

Reply:
Ok sounds good, let's keep it as a separate constraint then